ttter/accounts/views.py

```diff
@@ -39,3 +43,120 @@ def form_valid(self, form):
+
+
+class UserListView(LoginRequiredMixin, ListView):
+    template_name = "accounts/user_list.html"
+    model = User
+
+    def get_queryset(self):
+        return User.objects.values_list("username", flat=True)
+
+
+class UserPage(LoginRequiredMixin, TemplateView):
+    template_name = "accounts/user_page.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        request_user = self.request.user
+        user = get_object_or_404(User, username=self.kwargs.get("username"))
+        post_item = Tweet.objects.filter(user=user)
+        context["is_followed"] = FriendShip.objects.filter(
+            followee=request_user, follower=user
+        ).exists()
+        context["request_user"] = request_user
+        context["user"] = user
+        context["post_item"] = post_item
+        context["followee_count"] = user.followee.count()
+        context["follower_count"] = user.follower.count()
+        return context
+
+
+class FollowerListView(LoginRequiredMixin, TemplateView):
+    template_name = "accounts/follower_list.html"
+
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs)
+        self.user = get_object_or_404(User, username=self.kwargs["username"])
+
+        ctx["followers"] = [
+            follower.followee
+            for follower in FriendShip.objects.select_related("follower").filter(
+                follower=self.user
+            )
+        ]
+        return ctx
+
+
+class FollowingListView(LoginRequiredMixin, TemplateView):
+    template_name = "accounts/following_list.html"
+
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs)
+        self.user = get_object_or_404(User, username=self.kwargs["username"])
+        ctx["followings"] = [
+            follower.follower
+            for follower in FriendShip.objects.select_related("followee").filter(
+                followee=self.user
+            )
+        ]
+        return ctx
+
+
+class FollowView(LoginRequiredMixin, TemplateView):
+    template_name = "accounts/follow.html"
+    model = FriendShip
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        self.follower = get_object_or_404(User, username=self.kwargs["username"])
+        self.followee = get_object_or_404(User, id=self.request.user.id)
+        context["follower"] = self.follower
+        if self.followee == self.follower:
+            messages.warning(self.request, "自分自身はフォローできません。")
+        elif FriendShip.objects.filter(
+            followee=self.followee, follower=self.follower
+        ).exists():
+            messages.warning(self.request, f"{self.follower.username}さんはすでにフォローしています。")
+        return context
+
+    def post(self, *args, **kwargs):
+        self.follower = get_object_or_404(User, username=self.kwargs["username"])
+        self.followee = get_object_or_404(User, id=self.request.user.id)
```

Comment:
```suggestion
        follower = get_object_or_404(User, username=self.kwargs["username"])
        followee = get_object_or_404(User, id=self.request.user.id)
```
メソッド内で定義した変数にselfをつける必要ないです。UnfollowViewの方も同様です

Reply:
了解です！